core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -261,17 +397,14 @@ public long getOffsetInPage(long pagePlusOffsetAddress) {
    * value can be used to detect memory leaks.
    */
   public long cleanUpAllAllocatedMemory() {
-    long freedBytes = 0;
-    for (MemoryBlock page : pageTable) {
-      if (page != null) {
-        freedBytes += page.size();
-        freePage(page);
+    synchronized (this) {
+      Arrays.fill(pageTable, null);
+      for (MemoryConsumer c: consumers.keySet()) {
+        logger.warn("leak " + Utils.bytesToString(consumers.get(c)) + " memory from " + c);
```

Comment:
Just to clarify: is the memory leaked by these consumers also accounted for in the value returned by `releaseAllExecutionMemoryForTask`? Just want to make sure that this method's return value doesn't under- or over-count leaked memory.